src/ipc/probe_support.c

```diff
@@ -6,10 +6,11 @@
 
 #include <ipc/info.h>
 #include <sof/fw-ready-metadata.h>
+#include <sof/common.h>
 #include <config.h>
```

Comment:
Something bad happened here, ipc shouldn't have dependency on `sof/*` headers, that's also the reason why you don't see short version of attribute packed in ipc structs defs, cos you don't want to have that stuff to start leaking into ABI.
We once had arch-dependant ABI headers what is bug ofc, we don't want to go there again - config.h shouldn't be there for sure, the same goes for `arch/*` headers, also because of the fact that `sof/*` can include arch and platforms stuff it also shouldn't go there. All `ipc` headers should be independent.

Reply:
I think we should ban `config.h` and all `sof/*` headers in `ipc` folder, as it is something what should work for any config on any platform and including that stuff may affect output.
We can make one exception for some new header like `gcc-compat.h` where we have these gcc-provided short versions of attributes, to make ipc headers definitions look like from kernel.

@lgirdwood @dbaluta what do you think?

Reply:
@jajanusz not sure about ```config.h```. It might be used in the future to select stuff from Kconfig.

Reply:
@dbaluta Even though we may be able to select or deselect stuff from config.h, the ABI shouldn't change. Maybe add functions to query available functionality? (that's also going to be useful for our future use cases anyway)

Reply:
@dbaluta The thing is that ABI should work for driver that supports all possible platforms/configurations. If you allow `config.h` in abi headers, you open possibility to make f.e. ipc definitions that are different for `imx8` and `imx8m`.

Reply:
And as @paulstelian97 wrote, even if there are differencies they should be dynamically reported as some platform capabilities and generic 'abi' itself should be pretty much static and stable.

Reply:
@dbaluta @paulstelian97 i see. lets ban the ``config.h`` and ``sof/*`` headers in ipc folder then.

Reply:
What about some compatibility header with GCC extensions for compilers that don't have them? - I'm talking about __packed, __section etc.

Reply:
@jajanusz AFAIK __section needs to be defined in GCC as well, possibly __packed too. So this header can be common among compilers as all of them require defining them anyway.

Reply:
Ah, I see they are declared in `include/linux/compiler_attributes.h`, I guess we need something similar, we shouldn't include whole sof/common in uapi headers.

Reply:
I pushed new version with introduced sof/compiler_attributes.h